src/Symfony/Bundle/SecurityBundle/DataCollector/SecurityDataCollector.php

```diff
@@ -123,6 +128,17 @@ public function collect(Request $request, Response $response, \Exception $except
             $this->data['voter_strategy'] = 'unknown';
             $this->data['voters'] = array();
         }
+
+        // collect firewall context informations
+        if ($this->firewallMap instanceof ContextAwareFirewallMapInterface) {
+            $firewallContext = $this->firewallMap->getContext($request);
+
+            if (null !== $firewallContext) {
+                $this->data['firewall'] = $firewallContext->getConfig();
+            }
+        } else {
+            $this->data['firewall'] = array();
```

Comment:
So `$this->data['firewall']` may either contain a `FirewallConfig` instance or an empty array if unavailable ? It's quite strange, isn't it ?
What about not setting it at all if unavailable, and return `false` in `getFirewall()` ?

Twig checks will also make more sense: 

```diff
- {% if collector.firewall|default([]) is not empty %}
+ {% if collector.firewall %}
```

What do you think ?